packages/gatsby-source-wordpress/src/normalize.js

```diff
@@ -360,12 +362,21 @@ exports.searchReplaceContentUrls = function({
   })
 }
 
-exports.mapEntitiesToMedia = entities => {
+exports.mapEntitiesToMedia = (entities, _siteURL) => {
   const media = entities.filter(e => e.__type === `wordpress__wp_media`)
 
   return entities.map(e => {
     // Map featured_media to its media node
 
+    // Replace media URL if rely on localhost, when _siteURL isn't
+    if (
+      e.source_url &&
+      e.source_url.indexOf(`localhost`) !== -1 &&
+      _siteURL.indexOf(`localhost`) === -1
```

Comment:
This is problematic - checking if localhost is used (IMO) seems to handle very specific usecase (docker).

We also can't switch every domain to use one used for API url, as there are some wordpress plugin that push images to CDNs.

What about this - in part of code that does discover routes, we could try to figure out what replacement is needed (based on passed API url and self link reported by main `/wp-json` endpoint) and use that to perform any url swaps?

Reply:
Agree, better to rethink this part and move to another PR, at least it is not related to main PR idea. But it works pretty well without any issues, anyway it weird situation when _siteUrl goes to custom domain and self links in response from custom domain get us to localhost.

Reply:
If we could drop this change from this PR and move it separate one, we could merge the https://wordpress.org/plugins/wp-rest-api-v2-menus/ support faster. I'll go ahead and verify just the menu rest api part

Reply:
Also another idea - we have this `searchReplaceContentUrls` plugin option that potentially would cover this case (but not as automatically as solutions discussed here). It would be good idea to test that one out (one problem I see is that search & replace is run after media mapping + downloading (currently), so we would need to move it before those (tho maybe there is reason for doing this way).

Reply:
Good idea. I'll try to do it. But for now i think we can remove this part and if everything else is ok - accept this PR. As improvement it would be good to update docs in source-wp plugin